model/host/hostutil.go

```diff
@@ -66,18 +68,49 @@ func ChmodCommandWithSudo(ctx context.Context, script string, sudo bool) *exec.C
 	return exec.CommandContext(ctx, args[0], args[1:]...)
 }
 
-func (h *Host) CurlCommand(url string) string {
-	return fmt.Sprintf("cd ~ && curl -LO '%s/clients/%s' && chmod +x %s",
-		strings.TrimRight(url, "/"),
-		h.Distro.ExecutableSubPath(),
-		h.Distro.BinaryName())
+// CurlCommand returns the command to curl the evergreen client.
+func (h *Host) CurlCommand(settings *evergreen.Settings) string {
+	return h.CurlCommandWithRetry(settings, 0, 0)
+}
+
+// CurlCommandWithRetry is the same as CurlCommand but retries the request if
+// numRetries and maxRetrySecs are non-zero.
+func (h *Host) CurlCommandWithRetry(settings *evergreen.Settings, numRetries, maxRetrySecs int) string {
+	var retryArgs string
+	if numRetries != 0 && maxRetrySecs != 0 {
+		retryArgs = " " + curlRetryArgs(numRetries, maxRetrySecs)
+	}
+	return fmt.Sprintf("cd ~ && curl -LO '%s'%s && chmod +x %s",
+		h.ClientURL(settings),
+		retryArgs,
+		h.Distro.BinaryName(),
+	)
+}
+
+// Constants representing default curl retry arguments.
+const (
+	CurlDefaultNumRetries = 10
+	CurlDefaultMaxSecs    = 100
+)
+
+func curlRetryArgs(numRetries, maxSecs int) string {
+	return fmt.Sprintf("--retry=%d --retry-max-time=%d", numRetries, maxSecs)
+}
+
+// ClientURL returns the URL used to get the latest Evergreen client version.
+func (h *Host) ClientURL(settings *evergreen.Settings) string {
+	return fmt.Sprintf("%s/%s/%s",
+		strings.TrimRight(settings.Ui.Url, "/"),
```

Comment:
`TrimSuffix` here and the line below. `TrimRight` will have confusing behavior for multiple-character cutsets, so it's better to use `TrimSuffix` to be explicit about what you want, even though this cutset is single character. `TrimRight` will also cut repeat characters, which although fine here, can also be confusing.